### ISSUE TITLE
Add option to allow returning nullable from methods in Jsr305AnnotationCheck

### ISSUE THREAD

**Issue by vdaniloff** [sevntu-checkstyle/sevntu.checkstyle#887]
http://sevntu-checkstyle.github.io/sevntu.checkstyle/apidocs/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.html

/var/tmp $ javac Test.java

/var/tmp $ cat Test.java
```
package org.example;

import javax.annotation.Nullable;

public class Test {
    @Nullable // violation
    Object returnNullable() {
        return null;
    }
}
```

/var/tmp $ cat config.xml
```
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="Jsr305Annotations">
			<property name="packages" value="org.example"/>
		</module>
	</module>
</module>
```
For Linux users:

```/var/tmp $ java -classpath checkstyle-10.1-all.jar:sevntu-checks-1.41.0.jar  com.puppycrawl.tools.checkstyle.Main -c config.xml Test.java ```

For Windows users:

```C:\tmp> java -classpath checkstyle-10.1-all.jar;sevntu-checks-1.41.0.jar  com.puppycrawl.tools.checkstyle.Main -c config.xml Test.java ```

```
Starting audit...
[ERROR] C:\work\checkstyle\Test.java:6:5: @Nullable is not allowed on method return values. [Jsr305Annotations]
Audit done. 
Checkstyle ends with 1 errors.                                                                                    
```

Describe what you expect in detail.

Expected: 
A configuration option allowNullableReturnValue can be set, so that the following configuration could be used
```
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="Jsr305Annotations">
			<property name="packages" value="org.example"/>
                        <property name="allowNullableReturnValue" value="true"/>
		</module>
	</module>
</module>
```
And the result would be
```
Starting audit...            
Audit done. 
```

Motivation is in fact using `@Nullable` is quite common practice, though I realize `@CheckForNull` is recommended by most tools (including SonarQube)

**Comment by romani** [comment]
I am serious about https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by vdaniloff** [comment]
rewritten.

**Comment by romani** [comment]
@mbert, please review this request.
